src/components/button/Button.tsx

```diff
@@ -0,0 +1,91 @@
+import styled from "@emotion/styled";
+import Image from "next/image";
+import { useState, ButtonHTMLAttributes } from "react";
+import { getButtonStyles, getWidthStyles, setLoadingStyles } from "components/button/styles";
+
+export interface Props extends ButtonHTMLAttributes<HTMLButtonElement> {
+  variant?: "primary" | "secondary" | "text" | "icon" | "iconWithText"; // color types
+  widthType?: "default" | "fullWidth"; // default : hug
```

Comment:
오 좋아요!

다만 저희 디자인 상 `default` 도 많이 쓰지만 `fullWidth` 도 꽤 많이 쓸 것 같기도 하며, 현재 종류가 2가지 밖에 없어 `fullWidth` 라는 boolean props 를 받는 형식은 어떨까요?

```suggestion
  fullWidth?: boolean // default : false
```

이러면 사용처에서 바로 

```tsx
  <Button fullWidth /> { /* 길쭉버튼 */}
  <Button /> { /* 일반버튼 */}
```

등으로 쓸 수 있을 것 같습니다!

Reply:
Props를 boolean으로 하면 이게 편하더군요 ㅋㅋㅋ 전 좋다고 봅니당

Reply:
해결된 것 같아 Resolve 합니다~